framework/db/ActiveQuery.php

```diff
@@ -794,4 +805,64 @@ public function alias($alias)
         }
         return $this;
     }
+
+    /**
+     * Replace occurence of self::ALIAS_PLACEHOLDER by $aliasName into given conditions
+     * @param string|array $condition the given conditions
+     * @param string $aliasName
+     */
+    public function replaceAliasPlaceholder($condition, $aliasName)
+    {
+        if ((empty($condition)) or (is_int($condition))) {
+            return $condition;
+        } elseif (is_string($condition)) {
+            return str_replace(self::ALIAS_PLACEHOLDER, $aliasName, $condition);
+        } elseif (is_array($condition)) {
+            $conditionNew = [];
+            foreach ($condition as $key => $value) {
+                unset($condition[$key]);
```

Comment:
redundant since line 828 rewrites `$condition` anyway

Reply:
Thanks in removed it.